### PR TITLE
Setup wizard: fix logic error when detecting existing shipping zones.

### PR DESCRIPTION
**Pull request by jeffstieler**
To test:
* Remove all shipping zones
* Go to `/wp-admin/admin.php?page=wc-setup&step=shipping`
* Leave at least one zone enabled
* Click save
* Verify no zones are created
* Check out this branch
* Repeat steps
* Verify zones were created

### CODE REVIEW

**Comment by mikejolley** [comment]
@jeffstieler is the services logic below this not important?

**Comment by jeffstieler** [comment]
> @jeffstieler is the services logic below this not important?

It's important - this PR is just to fix an error in the logic for detecting existing shipping zones. (The thought being that we don't want to clobber and existing setup)

**Comment by mikejolley** [comment]
@jeffstieler Right but looking at the code in context, won't this ignore everything on the shipping step just displayed? Shouldn't the shipping step be hidden if no save logic runs?

**Comment by jeffstieler** [comment]
> Shouldn't the shipping step be hidden if no save logic runs?

We're saving the weight and dimension units no matter what. How about we hide the shipping zones portion if some already exist?

**Comment by mikejolley** [comment]
@jeffstieler Probably best otherwise you're showing UI which never gets used.

**Comment by DanReyLop** [comment]
@olaolusoga  Should we tweak the text in this case, or just remove it? The only fields would be the Weight/Length unit dropdowns. It's a corner case (you would need to create shipping zones *and* go through the wizard afterwards).

wizard

**Comment by olaolusoga** [comment]
We should switch copy to something along the lines of

"How would you like units on your store displayed?"

If it feels like overkill, we can remove it.

**Comment by DanReyLop** [comment]
> "How would you like units on your store displayed?"

Done, thanks!

File: includes/admin/class-wc-admin-setup-wizard.php [hunk 1]
```diff
@@ -753,44 +754,46 @@ public function wc_setup_shipping() {
...
-					</div>
-					<div class="wc-wizard-service-enable">
-						<span class="wc-wizard-service-toggle">
-							<input id="shipping_zones[intl][enabled]" type="checkbox" name="shipping_zones[intl][enabled]" value="yes" checked="checked" />
-							<label for="shipping_zones[intl][enabled]">
-						</span>
-					</div>
-				</li>
-			</ul>
+			<?php if ( empty( $existing_zones ) ) { ?>
```
Comment (jeffstieler) [inline includes/admin/class-wc-admin-setup-wizard.php]: Use PHP alternate syntax when mixing with HTML.

`<?php if ( empty( $existing_zones ) ) : ?>`
...
`<?php endif; ?>`

**Comment by jeffstieler** [comment]
One small comment - looks good otherwise. 👍

File: includes/admin/class-wc-admin-setup-wizard.php (hunk 1 above)
Comment (DanReyLop) [reply includes/admin/class-wc-admin-setup-wizard.php]: Changed in https://github.com/woocommerce/woocommerce/pull/16995/commits/f2d05be59b23b21c5d5f5281b9ae891b146e7e58

**Pull request merged**